eng/build.sh

```diff
@@ -110,9 +143,14 @@ while [[ $# > 0 ]]; do
 
      -subset|-s)
       if [ -z ${2+x} ]; then
-        arguments="$arguments /p:Subset=help"
-        shift 1
+        showSubsetHelp
+        exit 0
       else
+        passedSubset="$(echo "$2" | awk '{print tolower($0)}')"
```

Comment:
We also have detection on the first argument to parse it as a subset: https://github.com/dotnet/runtime/pull/37377/files#diff-75d32cbca8eddea951ce0484aa996d8fR130

So we also need to consider the case where people passes down:

`.\build.sh help`

Reply:
Just tested it. It works and shows the new subset message, but it shows the metrics. Let me add a handler for that case as well.